docs/synapse/userguides/ug011_storm_basics.rst

```diff
@@ -93,13 +64,158 @@ Working with Synapse data commonly involves three broad types of operations:
+
+To address these issues, Storm also supports what is known as **macro syntax.** Macro syntax acts as a sort of "shorthand" through techniques such as:
+
+* Replacing operators with equivalent intuitive symbols.
+* Allowing the omission of explicit operator names or parameters where there is an obvious default value.
+
+The macro syntax is meant to be both more efficient (requiring fewer keystrokes) and more intuitive, a "data language" for asking questions of the data as opposed to a programming language for retrieving data from a data store.
+
+While not every operator has a macro syntax equivalent, the most commonly used operators have been implemented both ways. When Storm macro syntax is used at the CLI, Synapse automatically "translates" the macro syntax to the equivalent operator syntax in order to execute the requested query.
+
+Two examples – one simple, one more complex – illustrate the differences between the two.
+
+*Example 1*
+
+The most basic Storm query simply lifts (retrieves) a single node (such as the domain ``woot.com``) using the ``lift()`` operator:
+
+``cli> ask lift(inet:fqdn,woot.com)``
+
+The same query can be executed as follows using macro syntax:
+
+``cli> ask inet:fqdn=woot.com``
+
+Note that in macro syntax, the ``lift()`` operator – the most fundamental Storm operator – is eliminated entirely; macro syntax assumes you want to retrieve (lift) nodes unless you specify otherwise. Similarly, instead of entering comma-separated parameters as input to the operator, macro syntax supports the use of the simple ``<form>=<valu>`` pattern to ask about the node in question.
+
+*Example 2*
+
+The usefulness of macro syntax is even more apparent with longer, more complex queries. Storm allows users to chain operators together to lift a set of nodes and perform a series of additional filter and pivot operations that follow a line of analysis across the data.
+
+In the knowledge domain of cyber threat data, there is a common analytical workflow used to research potentially malicious infrastructure. This line of analysis takes a set of “known bad” domains (for example, those associated with a known threat cluster), identifies the IP addresses those domains have resolved to, excludes some potentially irrelevant IPs, and then identifies other domains that have resolved to those IPs. Domains that resolved to the same IP address(es) as the “known bad” domains during the same time period may be associated with the same threat.
+
+The full query for this line of analytical reasoning using operator syntax would be::
+
+  cli> ask lift(inet:fqdn,by=tag,tc.t12) pivot(inet:dns:a:fqdn,inet:fqdn) 
+    pivot(inet:ipv4,inet:dns:a:ipv4) -#anon.tor -#anon.vpn 
+    pivot(inet:dns:a:ipv4,inet:ipv4) pivot(inet:fqdn,inet:dns:a:fqdn)
+
+The same query using macro syntax would be::
+
+  cli> ask inet:fqdn*tag=tc.t12 -> inet:dns:a:fqdn inet:dns:a:ipv4 -> inet:ipv4 -#anon.tor -#anon.vpn
```

Comment:
Hmm. Good point. I'm torn; it would be "nice to" show "look how much shorter macro syntax is"! But technically I can also shorten the operator syntax for the pivot() operators, by removing the "from" parameter where the "from" is the form of the node: 

``
ask lift(inet:fqdn,by=tag,tc.t12) pivot(inet:dns:a:fqdn,inet:fqdn)
``

could be:

``
ask lift(inet:fqdn,by=tag,tc.t12) pivot(inet:dns:a:fqdn,)
``

I think if we use the "full" operator syntax (for clarity) we should keep the "full"(ish) macro syntax (since this is a high-level example) and get more into the details for the individual operators and / or tutorial.